### PR TITLE
feat(docs): Update multi-version build to include API landing pages for each version

### PR DESCRIPTION
**Pull request by Josmithr**
Updates the website build to generate more user-friendly landing pages for the root of each version of the API docs. Currently, only our "V1" docs are live, but this PR adds a landing page for the "V2" docs, currently marked as a draft. Once we are ready to enable the V2 docs, we can remove the draft flag to enable it.

Before:
image

After:
image

V2 landing page:
image

### CODE REVIEW

File: docs/data/versions.json [hunk 1]
```diff
@@ -1,11 +1,12 @@
 {
 	"params": {
-		"currentVersion": "v1",
+		"currentVersion": "v2",
```
Comment (Josmithr) [inline docs/data/versions.json:3]: Don't check these changes in

File: docs/themes/thxvscode/layouts/apis/aggPackage.html [hunk 2]
```diff
@@ -3,22 +3,30 @@
...
-
-                {{ $packagePages := where .Site.AllPages "Params.kind" "Package" }}
-                {{ $topPackage := index (where $packagePages "Params.package" $.Params.package) 0 }}
-
                 <div>
-                    {{ $topPackage.Params.Summary | markdownify }}
+                    {{ .Summary | markdownify }}
                 </div>
 
-                <h3>List of bundled packages</h3>
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/apis/aggPackage.html:20]: This text was specific to the `fluid-framework` package use-case, and wasn't particularly meaningful. Now that we're using this template more generally, it has been removed.

File: docs/themes/thxvscode/layouts/apis/aggPackage.html [hunk 3]
```diff
@@ -28,13 +36,19 @@ <h3>List of bundled packages</h3>
...
+                        <!-- Iterate through specified package name strings -->
+                            {{ range $children }}
+                            <!-- Find and iterate over the actual corresponding package documents of matching version
+                                (we should only find 1 match)-->
+                            {{ range where $packagesWithMatchingVersion "Params.package" . }} 
+                                <tr>
+                                    <td><a href="{{ .RelPermalink | safeURL }}" title="{{ .Title }}">{{ .Title }}</a></td>
+                                    <td>{{ truncate 170 (.Params.Summary | markdownify) }}</td>
+                                </tr>
+                            {{ else }}
```
Comment (Josmithr) [inline docs/themes/thxvscode/layouts/apis/aggPackage.html:48]: Previously, this logic failed silently if we didn't find a match. Now we will log an error if we don't find one.

File: docs/_replacements/docs/api/v1/fluid-framework/index.md [hunk 4]
```diff
@@ -4,6 +4,7 @@ kind: "Package"
 type: apis
 package: "fluid-framework"
 unscopedPackageName: "fluid-framework"
+version: "v1"
```
Comment (Josmithr) [inline docs/_replacements/docs/api/v1/fluid-framework/index.md:7]: Tells the template to link to the v1 docs specifically

File: docs/api-markdown-documenter/render-api-documentation.js [hunk 5]
```diff
@@ -127,14 +136,6 @@ async function renderApiDocumentation(inputDir, outputDir, uriRootDir, apiVersio
 			let filePath = path.join(outputDir, `${document.documentPath}.md`);
 
 			try {
-				// Hugo uses a special file-naming syntax to represent documents with "child" documents in the same directory.
```
Comment (Josmithr) [inline docs/api-markdown-documenter/render-api-documentation.js:130]: The model file was the only "index.md" file, so now that we're not rendering it at all, this handling can be removed.

File: docs/_replacements/docs/api/v2/_index.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+title: "Fluid Framework V2 API Overview"
+type: apis
+version: "v2"
+area: "libraries"
+children:
+  - 'fluid-framework'
+  - '@fluidframework/azure-client'
+  - '@fluid-experimental/odsp-client'
+layout: aggPackage
+draft: false
```
Comment (Josmithr) [inline docs/_replacements/docs/api/v2/_index.md:11]:
```suggestion
draft: true
```

**Review by chentong7 — APPROVED** [review]

**Pull request merged**